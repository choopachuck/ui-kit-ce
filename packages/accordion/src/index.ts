export * from './Accordion'
export * from './AccordionItem'
export type {
  AccordionIconProps,
  AccordionComponentsConfig,
} from './components'
