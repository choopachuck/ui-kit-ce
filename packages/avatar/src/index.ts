export * from './Avatar'
export * from './AvatarGroup'
export * from './types'
export type {
  AvatarGroupExtraProps,
  AvatarGroupComponentsConfig,
} from './components'
export * from './classes'
export * from './constants'
