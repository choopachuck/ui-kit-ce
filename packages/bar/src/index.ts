export * from './Bar'
export * from './BarButton'
export * from './BarDate'
export * from './BarDivider'
export * from './BarDropdown'
export * from './BarDropdownItem'
export * from './BarItem'
export * from './BarMenuItem'
export * from './BarSearch'
export * from './BarSelect'
export * from './constants'
export * from './interfaces'
export * from './SubBar'
