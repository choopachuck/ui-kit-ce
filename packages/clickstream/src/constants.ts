export const CLICK_STREAM_EVENTS = ['click', 'change'] as const

export const CLICK_STREAM_INACTIVITY_TIME = 30000

export const CLICK_STREAM_BATCH_SIZE = 10

// Нотация нарушена для обратной совместимости с @sbol/clickstream-agent
export const CONNECTION_TYPES = {
  bluetooth: 'BLUETOOTH',
  cellular: 'CELLULAR',
  ethernet: 'ETHERNET',
  wifi: 'WIFI',
  mixed: 'UNKNOWN',
  none: 'UNKNOWN',
  wimax: 'UNKNOWN',
  other: 'UNKNOWN',
  unknown: 'UNKNOWN',
} as const

export const CLICK_STREAM_GENERIC_ATTRIBUTES = {
  OFF: 'data-click-stream-off',
} as const

export const ALIGNMENT = 10

export const MINUTES_IN_HOUR = 60

export const MS_IN_SECOND = 1000

export const SECONDS_IN_MINUTE = 60

export const MS_IN_MINUTE = SECONDS_IN_MINUTE * MS_IN_SECOND
