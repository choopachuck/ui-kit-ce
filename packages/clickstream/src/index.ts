export * from './ClickStreamProvider'
export type {
  ClickStreamBaseData,
  ClickStreamContextProps,
  ClickStreamElementProperty,
  ClickStreamElementType,
  ClickStreamEventBatch,
  ClickStreamEventBatchItem,
  ClickStreamEventData,
  ClickStreamEventMeta,
  ClickStreamProviderProps,
} from './types'
