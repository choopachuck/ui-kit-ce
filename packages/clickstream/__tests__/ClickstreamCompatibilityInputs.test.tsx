/*--------------------------------------------------------------------------------------------------
 * ------- Тестирование совместимости компонентов Input и Select с провайдером ClickStream ---------
 * -------------------------------------------------------------------------------------------------
 */

import * as React from 'react'
import { render, waitFor } from '@testing-library/react'
import { ClickStreamComponentsTest } from '../examples/ClickStreamComponentsTest'
import * as ClickStreamUtils from '../src/utils'
import { mockGlobals } from './helpers'
import { userEvent } from '../../../jest/utils'

jest.useFakeTimers('modern').setSystemTime(new Date('2020-01-01'))
jest
  .spyOn(ClickStreamUtils, 'getISOTime')
  .mockImplementation(() => '2020-01-01T00:00:00.078-00:00')

beforeAll(() => {
  mockGlobals()
})

it('clickstream lib components compatibility: input-text', async () => {
  const handleBatch = jest.fn()
  const handleSendEvent = jest.fn()
  const expectedBatchCalls = 2

  const { getByTitle } = render(
    <ClickStreamComponentsTest
      batchSize={expectedBatchCalls}
      onSendEvent={handleSendEvent}
      onBatch={handleBatch}
    />
  )

  const input = getByTitle('input-text')

  await waitFor(() => userEvent.click(input), { timeout: 1 })
  userEvent.keyboard('new value')
  await waitFor(() => userEvent.tab(), { timeout: 1 })

  await waitFor(() =>
    expect(handleBatch).toHaveBeenCalledTimes(expectedBatchCalls)
  )
  await waitFor(() => expect(handleSendEvent).toMatchSnapshot())
})

it('clickstream lib components compatibility: input-number', async () => {
  const handleBatch = jest.fn()
  const handleSendEvent = jest.fn()
  const expectedBatchCalls = 2

  const { getByTitle } = render(
    <ClickStreamComponentsTest
      batchSize={expectedBatchCalls}
      onSendEvent={handleSendEvent}
      onBatch={handleBatch}
    />
  )

  const input = getByTitle('input-number')

  await waitFor(() => userEvent.click(input), { timeout: 1 })
  userEvent.keyboard('15')
  await waitFor(() => userEvent.tab(), { timeout: 1 })

  await waitFor(() =>
    expect(handleBatch).toHaveBeenCalledTimes(expectedBatchCalls)
  )
  await waitFor(() => expect(handleSendEvent).toMatchSnapshot())
})

it('clickstream lib components compatibility: input-password', async () => {
  const handleBatch = jest.fn()
  const handleSendEvent = jest.fn()
  const expectedBatchCalls = 2

  const { getByTitle } = render(
    <ClickStreamComponentsTest
      batchSize={expectedBatchCalls}
      onSendEvent={handleSendEvent}
      onBatch={handleBatch}
    />
  )

  const input = getByTitle('input-password')

  await waitFor(() => userEvent.click(input), { timeout: 1 })
  userEvent.keyboard('new password')
  await waitFor(() => userEvent.tab(), { timeout: 1 })

  await waitFor(() =>
    expect(handleBatch).toHaveBeenCalledTimes(expectedBatchCalls)
  )
  await waitFor(() => expect(handleSendEvent).toMatchSnapshot())
})

it('clickstream lib components compatibility: input clear button', async () => {
  const handleBatch = jest.fn()
  const handleSendEvent = jest.fn()
  const expectedBatchCalls = 2

  const { getByTitle } = render(
    <ClickStreamComponentsTest
      batchSize={expectedBatchCalls}
      onSendEvent={handleSendEvent}
      onBatch={handleBatch}
    />
  )
  const clearButton = getByTitle('input-text-clear-button')

  await waitFor(() => userEvent.click(clearButton), { timeout: 1 })

  await waitFor(() =>
    expect(handleBatch).toHaveBeenCalledTimes(expectedBatchCalls)
  )
  expect(handleSendEvent).toMatchSnapshot()
})

it('clickstream lib components compatibility: masked-input', async () => {
  const handleBatch = jest.fn()
  const handleSendEvent = jest.fn()
  const expectedBatchCalls = 4

  const { getByTitle } = render(
    <ClickStreamComponentsTest
      batchSize={expectedBatchCalls}
      onSendEvent={handleSendEvent}
      onBatch={handleBatch}
    />
  )

  const input = getByTitle('masked-input') as HTMLInputElement

  await waitFor(() => userEvent.click(input), { timeout: 1 })

  await userEvent.typeMasked(input, '21212')

  await waitFor(() => userEvent.tab(), { timeout: 1 })

  await waitFor(() => userEvent.click(input), { timeout: 1 })

  // userEvent производит лишние манипуляции в selectionRange у инпута и перемещает каретку всегда в конец маски;
  // нужно всегда вручную указывать актуальную позицию каретки
  input.setSelectionRange(11, 11)
  await waitFor(() => userEvent.keyboard('[Backspace]'), { timeout: 1 })
  await waitFor(() => userEvent.tab(), { timeout: 1 })

  await waitFor(() =>
    expect(handleBatch).toHaveBeenCalledTimes(expectedBatchCalls)
  )
  await waitFor(() => expect(handleSendEvent).toMatchSnapshot())
})

it('clickstream lib components compatibility: switch', async () => {
  const handleBatch = jest.fn()

  const { getByTitle } = render(
    <ClickStreamComponentsTest onBatch={handleBatch} />
  )

  userEvent.click(getByTitle('switch'))

  await waitFor(() => expect(handleBatch).toHaveBeenCalledTimes(2))
  expect(handleBatch).toMatchSnapshot()
})

it('clickstream lib components compatibility: select', async () => {
  const handleBatch = jest.fn()
  const handleSendEvent = jest.fn()
  const expectedBatchCalls = 3

  const { getByTitle, findAllByRole } = render(
    <ClickStreamComponentsTest
      batchSize={expectedBatchCalls}
      onSendEvent={handleSendEvent}
      onBatch={handleBatch}
    />
  )

  const select = getByTitle('select')

  await waitFor(() => userEvent.click(select), { timeout: 1 })
  const listItems = await findAllByRole('option')
  await waitFor(() => userEvent.click(listItems[1]), { timeout: 1 })

  await waitFor(() =>
    expect(handleBatch).toHaveBeenCalledTimes(expectedBatchCalls)
  )

  // Могут возникать артефакты в снепшотах теста, связанные с дата атрибутами у options в компоненте.
  // Запускайте тесты через yarn test:unit clickstream без указания названия теста
  await waitFor(() => expect(handleSendEvent).toMatchSnapshot())
})

it('clickstream lib components compatibility: checkbox', async () => {
  const handleBatch = jest.fn()
  const handleSendEvent = jest.fn()
  const expectedBatchCalls = 2

  const { getByTitle } = render(
    <ClickStreamComponentsTest
      batchSize={expectedBatchCalls}
      onSendEvent={handleSendEvent}
      onBatch={handleBatch}
    />
  )

  const input = getByTitle('checkbox') as HTMLInputElement

  await waitFor(() => userEvent.click(input), { timeout: 1 })

  await waitFor(() =>
    expect(handleBatch).toHaveBeenCalledTimes(expectedBatchCalls)
  )
  await waitFor(() => expect(handleSendEvent).toMatchSnapshot())
})

it('clickstream lib components compatibility: radio', async () => {
  const handleBatch = jest.fn()
  const handleSendEvent = jest.fn()
  const expectedBatchCalls = 2

  const { getByTitle } = render(
    <ClickStreamComponentsTest
      batchSize={expectedBatchCalls}
      onSendEvent={handleSendEvent}
      onBatch={handleBatch}
    />
  )

  const input = getByTitle('radio') as HTMLInputElement

  await waitFor(() => userEvent.click(input), { timeout: 1 })

  await waitFor(() =>
    expect(handleBatch).toHaveBeenCalledTimes(expectedBatchCalls)
  )
  await waitFor(() => expect(handleSendEvent).toMatchSnapshot())
})
