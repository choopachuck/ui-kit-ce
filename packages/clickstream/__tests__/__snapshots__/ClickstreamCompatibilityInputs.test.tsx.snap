// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clickstream lib components compatibility: checkbox 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "title",
                "value": "checkbox",
              },
              Object {
                "key": "class",
                "value": "input-0-2-205",
              },
              Object {
                "key": "type",
                "value": "checkbox",
              },
            ],
            "elementEvent": "click",
            "elementTag": "input",
            "elementType": "checkbox",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "true",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "title",
                "value": "checkbox",
              },
              Object {
                "key": "class",
                "value": "input-0-2-205",
              },
              Object {
                "key": "type",
                "value": "checkbox",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "checkbox",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "true",
          },
          "event": Event {
            "isTrusted": true,
          },
        },
      ],
      Object {
        "applicationLanguage": "ru",
        "browser": "Mozilla/5.0 (linux) AppleWebKit/537.36 (KHTML, like Gecko) jsdom/16.7.0",
        "platform": "WEB",
        "screenSize": "0x0",
        "systemLanguage": "ru",
        "timeStamp": "2020-01-01T00:00:00.078-00:00",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`clickstream lib components compatibility: input clear button 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "class",
                "value": "input-0-2-27 inputMedium-0-2-29",
              },
              Object {
                "key": "data-v-uik-input-type",
                "value": "text",
              },
              Object {
                "key": "value",
                "value": "",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "text",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "role",
                "value": "button",
              },
              Object {
                "key": "aria-disabled",
                "value": "false",
              },
              Object {
                "key": "class",
                "value": "clearButton-0-2-36",
              },
              Object {
                "key": "aria-label",
                "value": "clearButton",
              },
              Object {
                "key": "title",
                "value": "input-text-clear-button",
              },
            ],
            "elementEvent": "click",
            "elementTag": "div",
            "elementType": "div",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "clearButton",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
      ],
      Object {
        "applicationLanguage": "ru",
        "browser": "Mozilla/5.0 (linux) AppleWebKit/537.36 (KHTML, like Gecko) jsdom/16.7.0",
        "platform": "WEB",
        "screenSize": "0x0",
        "systemLanguage": "ru",
        "timeStamp": "2020-01-01T00:00:00.078-00:00",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`clickstream lib components compatibility: input-number 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "number",
              },
              Object {
                "key": "title",
                "value": "input-number",
              },
              Object {
                "key": "class",
                "value": "input-0-2-27 inputMedium-0-2-29",
              },
              Object {
                "key": "value",
                "value": "",
              },
            ],
            "elementEvent": "click",
            "elementTag": "input",
            "elementType": "text",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "number",
              },
              Object {
                "key": "title",
                "value": "input-number",
              },
              Object {
                "key": "class",
                "value": "input-0-2-27 inputMedium-0-2-29",
              },
              Object {
                "key": "value",
                "value": "15,00",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "text",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "15,00",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
      ],
      Object {
        "applicationLanguage": "ru",
        "browser": "Mozilla/5.0 (linux) AppleWebKit/537.36 (KHTML, like Gecko) jsdom/16.7.0",
        "platform": "WEB",
        "screenSize": "0x0",
        "systemLanguage": "ru",
        "timeStamp": "2020-01-01T00:00:00.078-00:00",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`clickstream lib components compatibility: input-password 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "title",
                "value": "input-password",
              },
              Object {
                "key": "type",
                "value": "password",
              },
              Object {
                "key": "class",
                "value": "input-0-2-27 password-0-2-40 inputMedium-0-2-29",
              },
              Object {
                "key": "data-v-uik-input-type",
                "value": "password",
              },
              Object {
                "key": "value",
                "value": "",
              },
            ],
            "elementEvent": "click",
            "elementTag": "input",
            "elementType": "password",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "title",
                "value": "input-password",
              },
              Object {
                "key": "type",
                "value": "password",
              },
              Object {
                "key": "class",
                "value": "input-0-2-27 password-0-2-40 inputMedium-0-2-29",
              },
              Object {
                "key": "data-v-uik-input-type",
                "value": "password",
              },
              Object {
                "key": "value",
                "value": "",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "password",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "new password",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
      ],
      Object {
        "applicationLanguage": "ru",
        "browser": "Mozilla/5.0 (linux) AppleWebKit/537.36 (KHTML, like Gecko) jsdom/16.7.0",
        "platform": "WEB",
        "screenSize": "0x0",
        "systemLanguage": "ru",
        "timeStamp": "2020-01-01T00:00:00.078-00:00",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`clickstream lib components compatibility: input-text 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "title",
                "value": "input-text",
              },
              Object {
                "key": "class",
                "value": "input-0-2-27 inputMedium-0-2-29",
              },
              Object {
                "key": "data-v-uik-input-type",
                "value": "text",
              },
              Object {
                "key": "value",
                "value": "",
              },
            ],
            "elementEvent": "click",
            "elementTag": "input",
            "elementType": "text",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "title",
                "value": "input-text",
              },
              Object {
                "key": "class",
                "value": "input-0-2-27 inputMedium-0-2-29",
              },
              Object {
                "key": "data-v-uik-input-type",
                "value": "text",
              },
              Object {
                "key": "value",
                "value": "",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "text",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "new value",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
      ],
      Object {
        "applicationLanguage": "ru",
        "browser": "Mozilla/5.0 (linux) AppleWebKit/537.36 (KHTML, like Gecko) jsdom/16.7.0",
        "platform": "WEB",
        "screenSize": "0x0",
        "systemLanguage": "ru",
        "timeStamp": "2020-01-01T00:00:00.078-00:00",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`clickstream lib components compatibility: masked-input 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "masked-input",
              },
              Object {
                "key": "title",
                "value": "masked-input",
              },
              Object {
                "key": "class",
                "value": "input-0-2-27 inputMedium-0-2-29",
              },
              Object {
                "key": "value",
                "value": "",
              },
            ],
            "elementEvent": "click",
            "elementTag": "input",
            "elementType": "text",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "masked-input",
              },
              Object {
                "key": "title",
                "value": "masked-input",
              },
              Object {
                "key": "class",
                "value": "input-0-2-27 inputMedium-0-2-29",
              },
              Object {
                "key": "value",
                "value": "+7 (212) 128-__-__",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "text",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "+7 (212) 128-__-__",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "masked-input",
              },
              Object {
                "key": "title",
                "value": "masked-input",
              },
              Object {
                "key": "class",
                "value": "input-0-2-27 inputMedium-0-2-29",
              },
              Object {
                "key": "value",
                "value": "+7 (212) 128-__-__",
              },
            ],
            "elementEvent": "click",
            "elementTag": "input",
            "elementType": "text",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "+7 (212) 128-__-__",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "masked-input",
              },
              Object {
                "key": "title",
                "value": "masked-input",
              },
              Object {
                "key": "class",
                "value": "input-0-2-27 inputMedium-0-2-29",
              },
              Object {
                "key": "value",
                "value": "+7 (212) 12_-__-__",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "text",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "+7 (212) 12_-__-__",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
      ],
      Object {
        "applicationLanguage": "ru",
        "browser": "Mozilla/5.0 (linux) AppleWebKit/537.36 (KHTML, like Gecko) jsdom/16.7.0",
        "platform": "WEB",
        "screenSize": "0x0",
        "systemLanguage": "ru",
        "timeStamp": "2020-01-01T00:00:00.078-00:00",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`clickstream lib components compatibility: radio 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "title",
                "value": "radio",
              },
              Object {
                "key": "type",
                "value": "radio",
              },
              Object {
                "key": "class",
                "value": "input-0-2-200",
              },
              Object {
                "key": "value",
                "value": "undefined",
              },
            ],
            "elementEvent": "click",
            "elementTag": "input",
            "elementType": "radio",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "true",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "title",
                "value": "radio",
              },
              Object {
                "key": "type",
                "value": "radio",
              },
              Object {
                "key": "class",
                "value": "input-0-2-200",
              },
              Object {
                "key": "value",
                "value": "undefined",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "radio",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "true",
          },
          "event": Event {
            "isTrusted": true,
          },
        },
      ],
      Object {
        "applicationLanguage": "ru",
        "browser": "Mozilla/5.0 (linux) AppleWebKit/537.36 (KHTML, like Gecko) jsdom/16.7.0",
        "platform": "WEB",
        "screenSize": "0x0",
        "systemLanguage": "ru",
        "timeStamp": "2020-01-01T00:00:00.078-00:00",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`clickstream lib components compatibility: select 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "title",
                "value": "select",
              },
              Object {
                "key": "role",
                "value": "combobox",
              },
              Object {
                "key": "aria-haspopup",
                "value": "listbox",
              },
              Object {
                "key": "aria-expanded",
                "value": "true",
              },
              Object {
                "key": "class",
                "value": "resetButton-0-2-41 button-0-2-68 medium-0-2-70 opened-0-2-72",
              },
              Object {
                "key": "type",
                "value": "button",
              },
              Object {
                "key": "aria-activedescendant",
                "value": "v-uik-140-option-",
              },
            ],
            "elementEvent": "click",
            "elementTag": "button",
            "elementType": "button",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Выберите значение",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "role",
                "value": "option",
              },
              Object {
                "key": "aria-selected",
                "value": "false",
              },
              Object {
                "key": "id",
                "value": "v-uik-140-option-1",
              },
              Object {
                "key": "class",
                "value": "listItem-0-2-211 option-0-2-56 medium-0-2-217 optionMedium-0-2-58",
              },
            ],
            "elementEvent": "click",
            "elementTag": "li",
            "elementType": "li",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Первое",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "select",
              },
              Object {
                "key": "type",
                "value": "hidden",
              },
              Object {
                "key": "value",
                "value": "1",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "hidden",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "1",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
      ],
      Object {
        "applicationLanguage": "ru",
        "browser": "Mozilla/5.0 (linux) AppleWebKit/537.36 (KHTML, like Gecko) jsdom/16.7.0",
        "platform": "WEB",
        "screenSize": "0x0",
        "systemLanguage": "ru",
        "timeStamp": "2020-01-01T00:00:00.078-00:00",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`clickstream lib components compatibility: switch 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      MouseEvent {
        "isTrusted": false,
      },
      Object {
        "batteryLevel": "0",
        "connectionType": "UNKNOWN",
        "elementAttributes": Array [
          Object {
            "key": "data-v-uik-input-type",
            "value": "switch",
          },
          Object {
            "key": "title",
            "value": "switch",
          },
          Object {
            "key": "type",
            "value": "checkbox",
          },
          Object {
            "key": "class",
            "value": "input-0-2-49",
          },
        ],
        "elementEvent": "click",
        "elementTag": "input",
        "elementType": "checkbox",
        "pageName": "",
        "timeStamp": "2020-01-01T00:00:00.078-00:00",
        "url": "http://localhost/",
        "value": "true",
      },
      Array [
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "switch",
              },
              Object {
                "key": "title",
                "value": "switch",
              },
              Object {
                "key": "type",
                "value": "checkbox",
              },
              Object {
                "key": "class",
                "value": "input-0-2-49",
              },
            ],
            "elementEvent": "click",
            "elementTag": "input",
            "elementType": "checkbox",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "true",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "switch",
              },
              Object {
                "key": "title",
                "value": "switch",
              },
              Object {
                "key": "type",
                "value": "checkbox",
              },
              Object {
                "key": "class",
                "value": "input-0-2-49",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "checkbox",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "true",
          },
          "event": Event {
            "isTrusted": true,
          },
        },
      ],
    ],
    Array [
      Event {
        "isTrusted": true,
      },
      Object {
        "batteryLevel": "0",
        "connectionType": "UNKNOWN",
        "elementAttributes": Array [
          Object {
            "key": "data-v-uik-input-type",
            "value": "switch",
          },
          Object {
            "key": "title",
            "value": "switch",
          },
          Object {
            "key": "type",
            "value": "checkbox",
          },
          Object {
            "key": "class",
            "value": "input-0-2-49",
          },
        ],
        "elementEvent": "change",
        "elementTag": "input",
        "elementType": "checkbox",
        "pageName": "",
        "timeStamp": "2020-01-01T00:00:00.078-00:00",
        "url": "http://localhost/",
        "value": "true",
      },
      Array [
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "switch",
              },
              Object {
                "key": "title",
                "value": "switch",
              },
              Object {
                "key": "type",
                "value": "checkbox",
              },
              Object {
                "key": "class",
                "value": "input-0-2-49",
              },
            ],
            "elementEvent": "click",
            "elementTag": "input",
            "elementType": "checkbox",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "true",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "switch",
              },
              Object {
                "key": "title",
                "value": "switch",
              },
              Object {
                "key": "type",
                "value": "checkbox",
              },
              Object {
                "key": "class",
                "value": "input-0-2-49",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "checkbox",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "true",
          },
          "event": Event {
            "isTrusted": true,
          },
        },
      ],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
